### PR TITLE
[scp/ssh tests]: switch fedora to alpine

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:
We switched out to fedora to unblock various CI's in https://github.com/kubevirt/kubevirt/pull/8410 but definitely should be able to do this with alpine
and save some resources.
Needs an alpine bump because we fixed cloud init with https://github.com/kubevirt/kubevirtci/pull/876.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by fossedihelm** [comment]
@akalenyu Seems like that at least vmi_istio tests don't really like new alpine image :/

**Comment by akalenyu** [comment]
> @akalenyu Seems like that at least vmi_istio tests don't really like new alpine image :/

Yep and my change to the alpine image is the only diff... I suspect `sshd` never started up properly on alpine up until my change was introduced

**Comment by akalenyu** [comment]
> @akalenyu Seems like that at least vmi_istio tests don't really like new alpine image :/

PTAL

File: tests/network/vmi_istio.go [hunk 1]
```diff
@@ -511,11 +512,18 @@ func createPasstVm(ports []v1.Port) *v1.VirtualMachineInstance {
...
 	)
 	return vmi
 }
 
+func enablePasswordAuth() string {
+	sedCmd := "sed -ri 's/PasswordAuthentication no/PasswordAuthentication yes/' /etc/ssh/sshd_config"
+
+	return "#!/bin/sh\n" + sedCmd + "\n"
+}
+
```
Comment (fossedihelm) [inline tests/network/vmi_istio.go:526]: For the future: this could be moved in some utility package

**Review by fossedihelm — APPROVED** [review]
Thanks @akalenyu! Just some feedback and advice for future improvements.
IMHO in `tests/scp.go` the creation of the vmi can be moved in the `BeforeEach` section.
````
By("injecting a SSH public key into a VMI")
vmi := libvmi.NewAlpineWithTestTooling(
	libvmi.WithCloudInitNoCloudUserData(renderUserDataWithKey(pub), false))
vmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
Expect(err).ToNot(HaveOccurred())

vmi = tests.WaitUntilVMIReady(vmi, console.LoginToAlpine)
````
avoiding some duplicate setup code. This can be addressed in a follow PR.
Thanks!

File: tests/network/vmi_istio.go [hunk 2]
```diff
@@ -259,7 +259,7 @@ var istioTests = func(vmType VmType) {
 					vmiIP := libnet.GetVmiPrimaryIPByFamily(vmi, k8sv1.IPv4Protocol)
 
 					Expect(
-						checkSSHConnection(bastionVMI, "fedora", vmiIP),
+						checkSSHConnection(bastionVMI, "doesntmatter", vmiIP),
```
[the remark this answers is not in the document]
Comment (akalenyu) [reply tests/network/vmi_istio.go:262]: Yep same

[the remark this answers is not in the document]
Comment (rmohr) [reply tests/network/vmi_istio.go:262]: looks like we don't have any test at all where it matters? Can we just remove the argument and just hardcode it inside the function then?

[the remark this answers is not in the document]
Comment (fossedihelm) [reply tests/network/vmi_istio.go:262]: Sounds reasonable. I hadn't noticed that it wasn't used anywhere else.

[the remark this answers is not in the document]
Comment (akalenyu) [reply tests/network/vmi_istio.go:262]: Sounds totally reasonable

**Comment by fossedihelm** [comment]
/hold
@akalenyu probably a missing `needs-rebase` label

**Pull request merged**